neighsyncd/neighsync.cpp

```diff
@@ -19,9 +19,21 @@ using namespace swss;
 
 NeighSync::NeighSync(RedisPipeline *pipelineAppDB, DBConnector *stateDb) :
     m_neighTable(pipelineAppDB, APP_NEIGH_TABLE_NAME),
-    m_stateNeighRestoreTable(stateDb, STATE_NEIGH_RESTORE_TABLE_NAME),
-    m_AppRestartAssist(pipelineAppDB, "neighsyncd", "swss", &m_neighTable, DEFAULT_NEIGHSYNC_WARMSTART_TIMER)
+    m_stateNeighRestoreTable(stateDb, STATE_NEIGH_RESTORE_TABLE_NAME)
 {
+    m_AppRestartAssist = new AppRestartAssist(pipelineAppDB, "neighsyncd", "swss", DEFAULT_NEIGHSYNC_WARMSTART_TIMER);
```

Comment:
Can please elaborate why we need the changes in neighsyncd and warmrestartAssist ?

Reply:
@arlakshm The AppRestartAssist class was handling the reconciliation logic only for a single app-db table. This class is enhanced to handle reconciliation logic for multiple app-db tables, if the user of this class is the producer for multiple app-db ables as in the case of NAT. Earlier neighsyncd is the producer for a single app-db table and passing the app-db table to AppRestartAssist. Since this class is enhanced now, neighsyncd code is modified to call the new APIs.